xarray/core/dataset.py

```diff
@@ -306,7 +306,7 @@ class Dataset(Mapping, ImplementsDatasetReduce, BaseDataObject,
     groupby_cls = groupby.DatasetGroupBy
 
     def __init__(self, data_vars=None, coords=None, attrs=None,
-                 compat='broadcast_equals'):
+                 compat='broadcast_equals', encoding=None):
```

Comment:
I think I'd like to keep it here. Often times, I end up batch constructing objects with something like `xr.Dataset(..., **kwargs)` where `kwargs = {..., 'encoding': {'unlimited_dims': ['time']}}`. 

Also, since we include `encoding` in the `DataArray` constructor, it seems appropriate to have it here as well.